### DISCUSSION TITLE
Go to next / previous image on click

### DISCUSSION THREAD

**Discussion by desmond206x** [damongolding/immich-kiosk#18] [Feature request]
I would like to interact with the slideshow. If i click on the right half of the screen, it would be nice to jump to the next image.
On the left the same, but with the previous images.

I don't know if it is possible, because now you need to log the image history, but it would be a nice feature.

**Comment by nicodemusjls** [comment]
I'd add that I'd like three touch/click zones. Left = previous, Middle = pause, Right = next

I have a touch screen monitor displaying this in my living room. When talking to someone about a photo and it suddenly changes, I'd love to be able to go back to a photo. Sometimes I'd also like to pause one photo for a while.

**Comment by radh21301** [comment]
I was looking for this exact thing! I don't use any filters (albums/people) -> so sometimes some random screenshots/memes come though :)

Would be lovely if I can just go to the next picture with a touch/click.

**Comment by redge76** [comment]
Oh yes a "next zone" to hide some embarassing picture when my mother comes by would be a must.
(yes I should pick the album I display more carefully)
It's the most requested feature !!

**Comment by damongolding** [comment]
Work in Progress of next image on click/touch.

https://github.com/user-attachments/assets/0da6a5cc-2e98-4547-bfd0-9c6396fa222f

**Comment by damongolding** [comment]
More work in progress 😎

https://github.com/user-attachments/assets/75c86542-5cdf-4fc2-a14d-6bf0c94a6e33

**Reply by redge76** [reply]
Looks great

**Comment by damongolding** [comment]
If anyone wants to try out the test build the docker image is damongolding/immich-kiosk-development:0.13.0-beta.1

**Comment by redge76** [comment]
Juste tested the beta en it works perfectly. I personnaly would prefer the following control zones but it's a question of taste.

![image](https://github.com/user-attachments/assets/fb3ef194-b209-48f8-846a-0210a205360a)
or
![image](https://github.com/user-attachments/assets/73dec2f1-3b4c-41d0-98f1-a34754e5d50d)

I've been using pixette https://pixette.app/ which has this zone layout and I think it feels better because usually you go left or right. the "control zone" is less used.

Plus if you have a vertical image the left of right zones are outside of the actual picture (image taken vertically with a smartphone displayed on a wide PC screen)
It looks like this
![image](https://github.com/user-attachments/assets/a9b9c7d6-014b-46a1-ba2a-6db62b86e143)
If you click on the side of the picture, you are still in the "control" zone.

**Reply by damongolding** [reply]
Good points!

Happy to change it if that's the consensus.

**Reply by damongolding** [reply]
![localhost_3000](https://github.com/user-attachments/assets/99ace1b2-991f-46ae-9742-362803f4425c)

Going to set this as the default.

**Comment by contagon** [comment]
Just gave the beta a shot and works great! Been waiting/hoping for this for quite a while.

I'm trying to use this with [dashboard screensaver](https://play.google.com/store/apps/details?id=chyzinr.webpagescreensaver&hl=en_US) on a Google TV and it's not quite flawless (yet). I'd like my remote left/right arrows to scroll through the pictures, but it currently highlights the control rectangles of left/center/right on the screen and then I can click on them and it'll scroll.

Would it be too difficult to make left and right arrow keys also cycle through images? I wonder/think if that would do the trick

**Reply by contagon** [reply]
Hmm, actually played around with it more and seems to already setup with arrow keys? Maybe I need to get in touch with the dashboard screensaver dev to figure this out...

**Reply by damongolding** [reply]
You're right, it does already work with the left/right/space keyboard events. As a guess, I think the app you're using will trigger a tab for the right button and shift+tab for the left. 

I think I might be able to catch those events. Normally I wouldn't want to do that for accessibility issues but I could add a config option to enable it?

**Reply by contagon** [reply]
Played with it some more, it does seem to be catching the arrow keys! So everything is working properly which is awesome. The only catch is it highlights the large rectangles that correspond to the left/center/right portions of the screen you can click (this confused me and is why I missed that the arrow keys were working!)

Is there anyway to make these not-selectable on the immich-kiosk side?

**Reply by damongolding** [reply]
I've removed all focus styling. This is what causes the highlights.

**Reply by contagon** [reply]
Works like a charm now, officially 100% as good as the default google photos screensaver

As a small bug report if you're not aware, when I go back to a previous photo the location is gone, and the date is January 01, 0001.

**Reply by damongolding** [reply]
Thanks for the tip! and well noticed, it is indeed a bug
